otherlibs/unix/getgroups.c

```diff
@@ -28,15 +29,22 @@
 
 CAMLprim value caml_unix_getgroups(value unit)
 {
-  gid_t gidset[NGROUPS_MAX];
+  gid_t* gidset;
+  long ngroups_max;
   int n;
   value res;
 
-  n = getgroups(NGROUPS_MAX, gidset);
-  if (n == -1) caml_uerror("getgroups", Nothing);
+  if ((ngroups_max = sysconf(_SC_NGROUPS_MAX)) == -1)
```

Comment:
```suggestion
  if ((ngroups_max = sysconf(_SC_NGROUPS_MAX) + 1) == -1)
```

> If the effective group ID of the process is returned with the supplementary group IDs, the value returned shall always be greater than or equal to one and less than or equal to the value of `{NGROUPS_MAX}+1`.
>
> ```c
> ngroups_max = sysconf(_SC_NGROUPS_MAX) + 1;
> group = (gid_t *)malloc(ngroups_max *sizeof(gid_t));
> ```

See [`getgroups`](https://pubs.opengroup.org/onlinepubs/9799919799/functions/getgroups.html) POSIX docs.

Reply:
ah good catch, thanks